src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -120,42 +119,26 @@ public IndexResolverReplacer(IndexNameExpressionResolver resolver, ClusterServic
         this.clusterInfoHolder = clusterInfoHolder;
     }
 
-    private static final boolean isAllWithNoRemote(final String... requestedPatterns) {
-
-        final List<String> patterns = requestedPatterns==null?null:Arrays.asList(requestedPatterns);
-
-        if(IndexNameExpressionResolver.isAllIndices(patterns)) {
-            return true;
-        }
-
-        if(patterns.size() == 1 && patterns.contains("*")) {
+    private static boolean isAllWithNoRemote(final String... requestedPatterns) {
+        final List<String> patterns = requestedPatterns == null ? null : Arrays.asList(requestedPatterns);
+        if (IndexNameExpressionResolver.isAllIndices(patterns)) {
             return true;
         }
-
-        if(new HashSet<String>(patterns).equals(NULL_SET)) {
-            return true;
+        if (patterns.size() == 1) {
+            final String pattern = patterns.iterator().next();
+            return pattern == null || pattern.equals("*");
```

Comment:
When (use case) `pattern` will be `null`?

Reply:
Not sure of the use case / scenario where it may be null but looks like the check was specifically put in place hence retained it. Can any changes around this be done in separate PR ?

Reply:
It is strange to get an array with a single element that is `null` instead of an empty array. As we are touching code here, I'd like to understand what is the use case before introducing changes.

Reply:
I don't want to introduce any logical changes here. The logic before and after the code change is the same. 

There may be other scenarios with other requests too and this check seems to be specifically put in place hence don't want to remove it 

Here is a use-case
- SingleShardRequest [indices](https://github.com/opendistro-for-elasticsearch/security/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java#L705) where index can be [nullable](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/support/single/shard/SingleShardRequest.java#L81) 

- Checking request [here](https://github.com/opendistro-for-elasticsearch/security/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java#L271) which can have a single null element in array.

Reply:
The code for `SingleShardRequest` looks buggy to me. If `SingleShardRequest` index is not `null`, it will add it twice to the array, if it is `null`, it creates an array with single element that is `null` instead of an empty array or `null`.

I don't suggest to fix `SingleShardRequest` code in this PR, but it will be better to fix `SingleShardRequest` first in a separate PR.

Reply:
Already raised this PR https://github.com/opendistro-for-elasticsearch/security/pull/896

Reply:
Likely that will be indication of another bug similar to one fixed in #896.